lib/ModalPresenter.tsx

```diff
@@ -4,31 +4,90 @@ import React, {
   useImperativeHandle,
   useRef,
   FC,
+  PropsWithChildren,
 } from "react";
 import { Animated, StyleSheet, View, ViewProps } from "react-native";
 import RootSiblings, { RootSiblingParent } from "react-native-root-siblings";
+import { defaultQueueName, ModalQueueElement, ModalQueuesManager } from './ModalQueue';
 
+type ModalPresenterRef = {
+  animatedOut: (completion?: () => void) => void;
+};
+
+/**
+ * Props provided by this libary to your modal component.
+ */
 export type ModalContentProps = {
-  dismiss: () => void;
+  /**
+   * Dismiss the modal imperatively. Calling this function again after the modal is dismissed has no effect.
+   */
+  dismiss: (onDismiss?: () => void) => void;
 };
 
+/**
+ * Options you can specify when you call `enqueueModal`.
+ */
+export type ModalOptions = { 
+  /**
+   * If `queueName` matches the name of an existing queue, it adds this modal to that queue,
+   * otherwise, it creates a new queue wth this name and adds the modal to it.
+   * 
+   * If you don't specify a name, the defaut queue managed by this library will be used.
+   * 
+   * [Learn how queue works.](https://github.com/whitespectre/rn-modal-presenter#presenting-modals-in-a-queue)
+   */
+  queueName?: string;
+  /**
+   * Get notified when the modal is later dismissed.
+   */
+  onDismiss?: () => void;
+ } & Omit<ModalQueueElement, 'present'>;
+
+/**
+ * Modal handler is returned by `showModal` and `enqueueModal` functions.
+ */
 export type ModalHandler = {
+  /**
+   * Same `dismiss` function as in `ModalContentProps`.
+   */
   dismiss: () => void;
```

Comment:
shouldn't this also have an `onDismiss?: () => void` type?

Reply:
good catch!

Reply:
and then this one should be the one with `dismiss: ModalDismissFunc` right?